src/main/java/duke/logic/Parser.java

```diff
@@ -0,0 +1,275 @@
+package duke.logic;
+
+import duke.command.*;
+import duke.exception.*;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * Parser deals with making sense of user input.
+ *
+ * @author totsukatomofumi
+ */
+public class Parser {
+    /** Task list to be passed into command constructors. */
+    private TaskList taskList;
+
+    /** HashMap to contain abbreviated month and their numerical representation as a string pairs. */
+    private static HashMap<String, String> months = new HashMap<>();
+
+    static {
+        String[] monthsArr = {"jan", "feb", "mar", "apr", "may", "jun", "jul", "aug", "sep", "oct", "nov", "dec"};
+        int dayNum = 1;
+        for (String key : monthsArr) {
+            months.put(key, "" + dayNum);
+            ++dayNum;
+        }
+    }
+
+    /**
+     * Constructor for the parser.
+     *
+     * @param taskList the task list to be passed into the construction of commands.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Parses the user input to the appropriate command called, if any.
+     *
+     * @param response the user input.
+     * @return the command based on the user input.
+     */
+    public Runnable parse(String response) {
+        //strip() to allow for any (unintentional) whitespaces before or after
+        response = response.strip();
+        String lresponse = response.toLowerCase();  //for any caps commands
```

Comment:
Should `lresponse` be renamed into something that isn't abbreviated such as `lowercaseResponse`? The `l` could've meant left instead of lowercase.